lib/tasks/sample_data/payment_method_factory.rb

```diff
@@ -0,0 +1,52 @@
+require "tasks/sample_data/addressing"
+require "tasks/sample_data/logging"
+
+class PaymentMethodFactory
+  include Logging
+  include Addressing
+
+  def create_samples(enterprises)
+    log "Creating payment methods:"
+    distributors = enterprises.select(&:is_distributor)
+    distributors.each do |enterprise|
+      create_payment_methods(enterprise)
+    end
+  end
+
+  private
+
+  def create_payment_methods(enterprise)
+    return if enterprise.payment_methods.present?
+    log "- #{enterprise.name}"
+    create_cash_method(enterprise)
+    create_card_method(enterprise)
+  end
+
+  def create_cash_method(enterprise)
+    create_payment_method(
+      enterprise,
+      "Cash on collection",
+      "Pay on collection!",
+      Spree::Calculator::FlatRate.new
+    )
+  end
+
+  def create_card_method(enterprise)
```

Comment:
At this point, it's clear to me that `enterprise` might be worth being an ivar. It's class state.